### ISSUE TITLE
Test `terra` integration and `get_grid_data`

### ISSUE THREAD

**Issue by wkearn** [TopoToolbox/topotoolboxr#14]
I would like to use the data in TopoToolbox/snapshot_data with a test that uses `terra` to load the various DEMs and `get_grid_data` to confirm important information like the cellsize and dimensions of the array. This is not so important to test directly, but it will be useful to have this infrastructure set up ahead of time for later tests.

There are a few different ways we might go about doing this:

1. Distributing the data with the package: put the `.tif` files in `inst/extdata` (see https://r-pkgs.org/data.html#sec-data-extdata)
2. Provide functions to download the data (we have something like this in pytopotoolbox)
3. Download the data in a GitHub Action (this is more or less what we do in libtopotoolbox)

(1) would be nice, but the data files are huge. It is probably a better idea to supply a small DEM with the package or to use `terra`'s example `elev.tif`, which is a DEM of Luxembourg, as a running example.

(2) requires some effort to set up the testing environment, but I think aiming for something like https://r-pkgs.org/data.html#sec-data-persistent would be the right idea.

(3) is a good fallback, but it is only useful for testing on CI. One would still need to set up a local testing environment manually.

My proposal is

1. Start by implementing a test that loads the Luxembourg example from `terra` (something like `r <- rast(system.file("ex/elev.tif", package="terra")` does the job) and then running our `get_grid_data` to verify that its cellsize and dimensions are as expected.
2. Add a download script that downloads the snapshot data archive (https://github.com/TopoToolbox/snapshot_data/releases/latest/download/snapshot_data.tar.gz), unzips it, and stores the data somewhere useful (`tests/testthat/fixtures` based on https://r-pkgs.org/data.html#sec-data-persistent).
3. Add the test from 1. for each of the DEMs in the snapshot data archive, but only run the test if the data are available.

Does this make sense, @FancyBirb? Do you think there is anything else that we need to make this happen?

**Comment by FancyBirb** [comment]
It does make sense. A test is necessary, but i am only sure that the first first proposal will work. The second one feels like the issue that we had with the integration of libtotopotoolbox and that external source are not welcome in Cran, but as you gave me some sources to look at, i will try my best to implement something like that.

**Comment by FancyBirb** [comment]
is it really important that terra::xres(r) == terra::yres(r) for gradient8 to work. Because of this, the elev.tif cannot be used for wrap_gradient8. It seems like the yres and xres are not matching

**Comment by wkearn** [comment]
It looks like they are approximately equal:

```R
> library(terra)
> r <- rast(system.file("ex/elev.tif", package="terra")
> terra::xres(r) - terra::yres(r)
[1] 3.469447e-18
```

Instead of testing equality, we should use `all.equal(terra::xres(r), terra::yres(r))` in that condition. This function accounts for floating point rounding errors, which is what you are seeing above.

It is important that the resolutions be approximately equal because a lot of our algorithms make the assumption that the resolutions are equal. You might have noticed, for example, that `gradient8` only takes a `float cellsize` rather than a `float cellsize[2]`. `cellsize` is the resolution of the grid, and the function assumes that the grid spacing in the two dimensions is the same.

**Comment by FancyBirb** [comment]
There is now another problem. I dont know how to do a projection, because elev.tif is in lon/lat and we need to change that, but i cant get anything out of the documentation : https://search.r-project.org/CRAN/refmans/terra/html/project.html

**Comment by wkearn** [comment]
Closed by #15
